Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -179,35 +181,71 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
                     break;
             }
 
-            IMixedRealityInputSource inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}");
-
-            GenericOpenVRController detectedController = null;
+            Type controllerType;
 
-            // Initialize the controller base on the detected type
             switch (CurrentControllerType)
             {
                 case SupportedControllerType.GenericOpenVR:
-                    detectedController = new GenericOpenVRController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(GenericOpenVRController));
+                    controllerType = typeof(GenericOpenVRController);
                     break;
                 case SupportedControllerType.ViveWand:
-                    detectedController = new ViveWandController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(ViveWandController));
+                    controllerType = typeof(ViveWandController);
                     break;
                 case SupportedControllerType.ViveKnuckles:
-                    detectedController = new ViveKnucklesController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(ViveKnucklesController));
+                    controllerType = typeof(GenericOpenVRController);
                     break;
                 case SupportedControllerType.OculusTouch:
-                    detectedController = new OculusTouchController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(OculusTouchController));
+                    controllerType = typeof(OculusTouchController);
                     break;
                 case SupportedControllerType.OculusRemote:
+                    controllerType = typeof(OculusRemoteController);
                     break;
+                default:
+                    Debug.LogError($"Unsupported controller type detected.");
+                    return null;
             }
 
+            var pointers = new List<IMixedRealityPointer>();
+
+            if (MixedRealityManager.HasActiveProfile &&
+                MixedRealityManager.Instance.ActiveProfile.IsInputSystemEnabled &&
+                MixedRealityManager.Instance.ActiveProfile.PointerProfile != null)
```

Comment:
I suppose we're assuming that all Open VR Controllers support pointing?

Reply:
good question. I have limited experience with their controllers

Reply:
@SimonDarksideJ would know, but he's on holiday

Reply:
OpenVR controllers only support pointing, e.g. one pose.
Testing however, will prove whether each needs a different offset pose for forward.  
They don't have a grip pose, which I proposed we may have to simulate later.

Reply:
So the nice thing is that pointers can choose to get their input data from the device's pose, or listen to input actions from that device, which makes choosing the source easy.